Source/Particles/SpeciesPhysicalProperties.H

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2020 Maxence Thevenet
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_SPECIESPHYSICALPROPERTIES_H_
+#define WARPX_SPECIESPHYSICALPROPERTIES_H_
+
+#include "WarpXConst.H"
+
+#include <AMReX_REAL.H>
+#include <AMReX_AmrCore.H>
+
+#include <map>
+#include <limits>
+
+enum struct PhysicalSpecies{unspecified=0, electron, positron, photon, hydrogen};
+
+static PhysicalSpecies from_string(std::string species)
+{
+    if( species=="unspecified" )
+        return PhysicalSpecies::unspecified;
+    if( species=="electron" )
+        return PhysicalSpecies::electron;
+    if( species=="positron" )
+        return PhysicalSpecies::positron;
+    if( species=="photon" )
+        return PhysicalSpecies::photon;
+    if( species=="hydrogen" )
+        return PhysicalSpecies::hydrogen;
+    amrex::Abort("unknown PhysicalSpecies");
+    return PhysicalSpecies::unspecified;
+}
+
+static amrex::Real get_charge (PhysicalSpecies ps)
+{
+    switch(ps) {
+    case PhysicalSpecies::unspecified:
+        return std::numeric_limits<amrex::Real>::quiet_NaN();
+    case PhysicalSpecies::electron:
+        return -PhysConst::q_e;
+    case PhysicalSpecies::positron:
+        return PhysConst::q_e;
+    case PhysicalSpecies::photon:
+        return std::numeric_limits<amrex::Real>::quiet_NaN();
```

Comment:
are we sure that we want `quiet_NaN` instead of zero for photons?

Reply:
You're right, I don't think we divide by this so it's fine to just put the to 0.